### PR TITLE
Fix update dependencies

### PR DESCRIPTION
**Pull request by eerhardt**
Fix update-dependencies by slimming it way down.

It can no longer be dependent on Cli.Utils, since that will affect which version of NuGet gets used.  It's better for this small console app to just depend on VersionTools.

We were getting these errors:

```
Unhandled Exception: System.IO.FileNotFoundException: Could not load file or assembly 'NuGet.Packaging.Core.Types, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The system cannot find the file specified.
at Microsoft.DotNet.VersionTools.Dependencies.DependencyBuildInfo..ctor(BuildInfo buildInfo, Boolean upgradeStableVersions, IEnumerable`1 disabledPackages)
at Microsoft.DotNet.Scripts.Program.<>c.<Main>b__1_0(BuildInfo buildInfo)
```

Which was caused by refactoring in NuGet.  VersionTools uses an older version, where the CLI has moved on to 4.3.0, which no longer has `NuGet.Packaging.Core.Types`.

I also did some clean up of some unused build code while I was looking in here.  Probably best reviewed commit-by-commit.

@ellismg @livarcocc

### CODE REVIEW

**Comment by ellismg** [comment]
Looks reasonable to me.

File: build/package/Installer.MSI.targets [hunk 1]
```diff
@@ -45,8 +45,7 @@
       <GenerateMsiVersion CommitCount="$(CommitCount)"
                           VersionMajor="$(VersionMajor)"
                           VersionMinor="$(VersionMinor)"
-                          VersionPatch="$(VersionPatch)"
-                          ReleaseSuffix="$(ReleaseSuffix)">
```
Comment (livarcocc) [inline build/package/Installer.MSI.targets]: Why is it we don't need the releasesuffix in the msi anymore?

**Comment by livarcocc** [comment]
@MattGertz for approval.

File: build/package/Installer.MSI.targets (hunk 1 above)
Comment (eerhardt) [inline build/package/Installer.MSI.targets]: Why is it this property wasn't being used?  IDK

Comment (eerhardt) [inline build/package/Installer.MSI.targets]: It does use CommitCount, but that might not give a 100% unique version across branches with the same major.minor.path-commitcount

Comment (eerhardt) [inline build/package/Installer.MSI.targets]: We weren't using ReleaseSuffix in 1.0.0-preview2.1 either:

https://github.com/dotnet/cli/blob/rel/1.0.0-preview2.1/build_projects/shared-build-targets-utils/Utils/Version.cs#L14-L37

**Comment by eerhardt** [comment]
With this change, we should start to see auto-PRs of new runtimes into the cli/master branch again.

**Pull request merged**